### DISCUSSION TITLE
Pull request merged doesn't start workflow on pull_request closed event

### DISCUSSION THREAD

**Discussion by korthout** [community/community#25737] [Actions]
<p>A Pull Request, merged through <code>gh pr merge</code> did not start a workflow that was listening on <code>pull_request</code> with type <code>closed</code>.</p>
<p>It seems like the pull_request closed event was not registered or somehow did not start the workflow that is listening on this event.</p>
<hr>
<p>I have <a href="https://github.com/korthout/backport-action-test/blob/dbc345ae2d713bfdcdce98c8e108090b80fab6c0/.github/workflows/backport-pr-closed.yml" rel="noopener nofollow ugc">a workflow</a> that should be triggered on <code>pull_request</code> <code>closed</code> events:</p>
<pre><code class="lang-auto">name: Backport on pull_request[closed]
on:
  pull_request:
    types: [closed]
jobs:
  backport:
    name: Backport
    runs-on: ubuntu-latest
....
</code></pre>
<p><a href="https://github.com/korthout/backport-action-test/blob/dbc345ae2d713bfdcdce98c8e108090b80fab6c0/.github/workflows/test.yml" rel="noopener nofollow ugc">Another workflow</a> can be run manually, which runs <a href="https://github.com/korthout/backport-action-test/blob/dbc345ae2d713bfdcdce98c8e108090b80fab6c0/test/case1.sh" rel="noopener nofollow ugc">a shell script</a> that creates and merges a pull request, and then cleans up the mess it created (delete the new branches and reverts the merge).</p>
<p>Merging the pull request, should then trigger the above mentioned workflow. However, this does not happen.</p>
<p>See:</p>
<ul>
<li>Test workflow was executed: <a href="https://github.com/korthout/backport-action-test/runs/4826743507" class="inline-onebox" rel="noopener nofollow ugc">Revert "case(1): merge pull request" · korthout/backport-action-test@632a714 · GitHub</a>
</li>
<li>Pull request was opened and merged: <a href="https://github.com/korthout/backport-action-test/pull/6" class="inline-onebox" rel="noopener nofollow ugc">Case(1): Add a changed line by github-actions[bot] · Pull Request #6 · korthout/backport-action-test · GitHub</a>
</li>
<li>No backport workflows run: <a href="https://github.com/korthout/backport-action-test/actions/workflows/backport-pr-closed.yml" class="inline-onebox" rel="noopener nofollow ugc">Actions · korthout/backport-action-test · GitHub</a>
</li>
</ul>
<p>Could it be that the time that the pull request is open is important for the closed event to be triggered? Because of the automation, the Create and the Merge happen practically immediately after each other. Or is there something else I’m missing?</p>
<p>It feels like a bug to me, because the pull request is clearly merged (and thus closed), so the pull_request closed event should’ve triggered the workflow.</p>

**Comment by korthout** [comment, marked as answer]
<p>Seems I did not read <a href="https://github.community/t/the-github-actions-guidepost/168114" class="inline-onebox">📍 The GitHub Actions Guidepost</a> well enough.</p>
<blockquote>
<p>events triggered by the <code>GITHUB_TOKEN</code> will not create a new workflow run</p>
</blockquote>
<p>Solution:</p>
<blockquote>
<p>If you would like to trigger a workflow from a workflow run, you can <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token" rel="noopener nofollow ugc">trigger the event using a personal access token</a></p>
</blockquote>

**Comment by mpboom** [comment]
<p>Thanks for sharing your solution with the community <a class="mention" href="/korthout">@korthout</a> !</p>
